### PR TITLE
(#52, #22) Fix No Trials Found functionality; prerender meta tags

### LINKED ISSUE #22: Enabler: Make redirection status available to prerenderers
# Title of Enabler

When the application redirects to another route, the resulting page should contain either:

**For temporary redirects:**
```html
<meta name="prerender-status-code" content="302">
<meta name="prerender-header" content="Location: https://www.google.com">
```

**For permanent redirects:**
```html
<meta name="prerender-status-code" content="301">
<meta name="prerender-header" content="Location: https://www.google.com">
```

In either case, the `prerender-header` tag should reflect the new URL and not Google.

This will allow a pre-renderer to detect the redirection and report the correct status code and URL.

> **ESTIMATE** 2
### Acceptance Criteria
The 302 redirects (which are used for No Trials Found) ACs can be found on #52.

For 301s (which are used for ugly to pretty URL name redirects):
```gherkin
Scenario: Page redirects to pretty URL if codes are given for Disease
  Given "trialListingPageType" is set to "Disease"
  And "pageTitle" is set to "{{disease_label}} Clinical Trials"
  Given the user navigates to "/C4872"
  Then the user is redirected to "/breast-cancer?redirect=true"
  And the page contains meta tags with the following names
    | name                  | content 				                           |
    | prerender-status-code | 301     			                                   |
    | prerender-header      | Location: http://localhost:3000/breast-cancer?redirect=true  |
```

### Notes

*Some complementary notes if necessary:*

* > Here goes a quote from an email
* Here goes whatever useful information can exist…
* More notes, or technical nuances/direction

### LINKED ISSUE #52: No Trials Found functionality and prerender status codes are incorrect
The current functionality of the No Trials Found is:
1) Go to /C3037, redirect to /notrials?p1=chronic-fatigue-syndrome with No Trials Found content and 302 prerender-status-code
2) Go to /chronic-fatigue-syndrome, redirect to /notrials?p1=chronic-fatigue-syndrome with No Trials Found content and 302 prerender-status-code
3) Go to /notrials?p1=C3037, get Page Not Found content with 404 prerender-status-code
4) Go to /notrials?p1=chronic-fatigue-syndrome, get Page Not Found content with 404 prerender-status-code
5) Go to /notrials?p1=C123456, get Page Not Found content with 404 prerender-status-code
6) Go to /notrials?p1=chicken, get Page Not Found content with 404 prerender-status-code

It should be:
1) Go to /C3037, redirect to /notrials?p1=chronic-fatigue-syndrome with No Trials Found content and 302 prerender-status-code
2) Go to /chronic-fatigue-syndrome, redirect to /notrials?p1=chronic-fatigue-syndrome with No Trials Found content and 302 prerender-status-code
3) Go to /notrials?p1=C3037, get No Trials Found content with 404 prerender-status-code
4) Go to /notrials?p1=chronic-fatigue-syndrome, get No Trials Found content with 404 prerender-status-code
5) Go to /notrials?p1=C123456, get Page Not Found content with 404 prerender-status-code
6) Go to /notrials?p1=chicken, get Page Not Found content with 404 prerender-status-code

> **ESTIMATE** 3

### Steps to reproduce the issue

1.  
2. 
3. 

### What's the expected result?
-

### What's the actual result?
-

### Additional details / screenshot
- Screenshot

### Related Tickets
* Issue #9999
* Issue #9999

### PR DESCRIPTION
**Pull request by sarinapadilla**
(#52) Fix No Trials Found functionality
Closes #52

* Update CTLViewsHoC with logic to handle /notrials route and prerender meta tags
* Update unit tests for CTLViewsHoC /notrials cases
* Update Disease to set correct state values on NoTrialsFound redirect
* Update unit tests for Disease
* Update NoTrialsFound to display prerender meta tags
* Update integration tests for NoTrialsFound

(#22) Prerender redirection status
Closes #22

* Update CTLViewsHoC to pass status for ugly to pretty URL redirects
* Update Disease component to display prerender meta tags if redirected
* Update Disease unit tests
* Update integration tests for ugly to pretty URL redirects

### CODE REVIEW

**Comment by sarinapadilla** [comment]
@VictoriaSunNIH @mworrest @alinai this is ready for testing. Here's a few example URLs for different scenarios.

1) Redirects to /notrials?p1={pretty-url-name}, shows No Trials Page with 302 prerender status
https://react-app-dev.cancer.gov/clinical-trials-listing-app/pr-54/C3037?cfg=0
https://react-app-dev.cancer.gov/clinical-trials-listing-app/pr-54/chronic-fatigue-syndrome?cfg=0

2) Redirect to /notrials?p1={pretty-url-name}, shows No Trials Page with 404 prerender status
https://react-app-dev.cancer.gov/clinical-trials-listing-app/pr-54/notrials?p1=C3037&cfg=0

3) Shows No Trials Page with 404 prerender status
https://react-app-dev.cancer.gov/clinical-trials-listing-app/pr-54/notrials?p1=chronic-fatigue-syndrome&cfg=0

4) Displays Page Not Found with 404 prerender status
https://react-app-dev.cancer.gov/clinical-trials-listing-app/pr-54/notrials?p1=chicken&cfg=0
https://react-app-dev.cancer.gov/clinical-trials-listing-app/pr-54/notrials?p1=C123456789&cfg=0

If you guys need more examples for diseases that return no trials (with one code, multiple codes, etc) I can find them for you!

**Comment by mworrest** [comment]
I can't speak for the prerender status but the redirects look accurate.

**Comment by alinai** [comment]
Looks good to me. Thanks @sarinapadilla!

**Comment by sarinapadilla** [comment]
@VictoriaSunNIH @mworrest @alinai @KateMashkinaNIH , this is ready for testing again! We have changed some requirements so I've updated the scenarios.

Redirects to /notrials?p1={code}, shows No Trials Page with 301 prerender status and prerender-header
https://react-app-dev.cancer.gov/clinical-trials-listing-app/pr-54/C3037?cfg=0

Redirects to /notrials?p1={pretty-url-name}, shows No Trials Page with 302 prerender status and prerender-header
https://react-app-dev.cancer.gov/clinical-trials-listing-app/pr-54/chronic-fatigue-syndrome?cfg=0

Shows No Trials Page with 404 prerender status and no prerender-header
https://react-app-dev.cancer.gov/clinical-trials-listing-app/pr-54/notrials?p1=C3037&cfg=0
https://react-app-dev.cancer.gov/clinical-trials-listing-app/pr-54/notrials?p1=chronic-fatigue-syndrome&cfg=0

Displays Page Not Found with 404 prerender status and no prerender-header
https://react-app-dev.cancer.gov/clinical-trials-listing-app/pr-54/notrials?p1=chicken&cfg=0
https://react-app-dev.cancer.gov/clinical-trials-listing-app/pr-54/notrials?p1=C123456789&cfg=0

**Comment by sarinapadilla** [comment]
Additionally, @KateMashkinaNIH, please test that the prerender meta tags are being populated correctly for ugly to pretty URL name redirects (#22), which is also part of this PR.

**Review by KateMashkinaNIH — CHANGES_REQUESTED** [review]

**Comment by KateMashkinaNIH** [comment]
tested various redirects - meta tags looks good

**Comment by alinai** [comment]
Looks good to me. Some examples of diseases with multiple C-codes provided by @sarinapadilla ,
C-Code = ["C43625", "C114993"], pretty url = "pleomorphic-hepatocellular-cancer"
C-Code = ["C4661", "C115253"], pretty url = "primary-melanocytic-lesion-of-meninges"

Thanks @sarinapadilla!

**Comment by VictoriaSunNIH** [comment]
Looks good to me too - so it sounds like this passes both QA and UX review - I'm updating labels.

File: cypress/integration/PageNotFound/NoTrialsRoutePageNotFound.feature [hunk 1]
```diff
@@ -1,8 +1,18 @@
...
 		Given "trialListingPageType" is set to "Disease"
-		When the user navigates to "/notrials?p1=C3037"
+		When user navigates to non-existent page "/notrials?p1=c123455&cfg=0"
+		Then page title on error page is "Page Not Found"
+    And the text "We can't find the page you're looking for." appears on the page
+    And the link "homepage" to "https://www.cancer.gov" appears on the page
+    And the link "cancer type" to "https://www.cancer.gov/types" appears on the page
+    And the link "Get in touch" to "https://www.cancer.gov/contact" appears on the page
+    And the search bar appears below
+
```
[the remark this answers is not in the document]
Comment (sarinapadilla) [reply cypress/integration/PageNotFound/NoTrialsRoutePageNotFound.feature:15]: We don't want the noindex tag on the Page Not Found, but you're right that we should be checking for the prerender status code tag! It's been fixed

**Review by KateMashkinaNIH — APPROVED** [review]

**Review by arcepaul — APPROVED** [review]

**Pull request merged**